### ISSUE TITLE
Apptainer application guide example on DMTCP checkpointing does not work.

### ISSUE THREAD

**Issue by Daymorn** [apptainer/apptainer#664]
```
apptainer version 1.1.0.rc.2
# cat /etc/redhat-release 
Rocky Linux release 8.6 (Green Obsidian)
# dmtcp_command --version
dmtcp_command (DMTCP) 3.0.0
```
When going through the guide at https://apptainer.org/docs/user/main/checkpoint.html I am not able to get the provided example to work. I have tried other containers that fail in either the same way, or don't make it as far.  Below is my attempt at the Apptainer example.

```
[user@smnode01 ~]$ apptainer instance start --dmtcp-launch example-checkpoint server.sif server 8888
INFO:    underlay of /usr/bin/dmtcp_command required more than 50 (445) bind mounts

INFO:    instance started successfully

[user@smnode01 ~]$ apptainer instance list
INSTANCE NAME    PID       IP    IMAGE
server           888957          /home/jgibbens/server.sif

[user@smnode01 ~]$ dmtcp_command -p $(cat ~/.apptainer/checkpoint/dmtcp/example-checkpoint/coord.port) -s
Coordinator:
  Host: localhost
  Port: 34299
Status...
  NUM_PEERS=1
  RUNNING=no
  CKPT_INTERVAL=0 (checkpoint manually)

# I added a fmt.Printf to cmd/internal/cli/checkpoint.go so I could see the command apptainer was using to launch dmctp
[user@smnode01 ~]$ apptainer checkpoint instance server
INFO:    Using checkpoint "example-checkpoint"
[/.singularity.d/actions/exec dmtcp_command --coord-port 34299 -c]
Error, computation not in running state.  Either a checkpoint is
 currently happening or there are no connected processes.
```

### How did you install Apptainer
git cloned and built RPMs
### How did you install DMTCP
```
dnf install libstdc++-static
cd /opt/act/src
git clone https://github.com/dmtcp/dmtcp.git
cd dmtcp/
./configure --enable-static-libstdcxx
make install
```

I am able to use dmtcp_launch and ./dmtcp_restart_script.sh on simple bash scripts that output incrementing numbers to an output file. The errors I am getting above are definitely on the DMTCP side, I just don't know if the core problem is with Apptainer or DMTCP, or if there is anything I can do about it at this point since this is experimental.

**Comment by GodloveD** [comment]
I am seeing the same error.  Can help to reproduce if that is useful.

**Comment by DrDaveD** [comment]
@ikaneshiro This is waiting on you.

**Comment by ikaneshiro** [comment]
Hi @Daymorn,

Without seeing full debug output of your container startup (apptainer -d), I suspect the dmtcp libraries are not within your linker path and are not being brought into your containers for use by the dmtcp binaries within your container.

This is likely the issue if you get no output when running: `ldconfig -v | grep dmtcp`

To fix this you will need to place the dmtcp libraries (usually built into /usr/local/lib/dmtcp/) inside a path your linker will pick up, or add their path to your linker paths. Once you do this you need to refresh your cache with `ldconfig`.

**Comment by DrDaveD** [comment]
I had the exact same problem going through the example.  I got it to work by these commands as root
```
# echo /usr/local/lib/dmtcp >/etc/ld.so.conf.d/dmtcp.conf
# ldconfig
```

**Comment by DrDaveD** [comment]
Alternatively if you don't have root access you could set `LD_LIBRARY_PATH=/usr/local/lib/dmtcp` and then use the [workaround](https://github.com/apptainer/singularity/issues/5759#issuecomment-919523970) I posted for another issue.
